inginious/frontend/plugins/grader_generator/pages/grader.py

```diff
@@ -31,27 +31,30 @@ def on_task_editor_submit(course, taskid, task_data, task_fs):
     # Create form object
     task_data["generate_grader"] = "generate_grader" in task_data
 
-    if task_data['generate_grader']:
-        if task_data['environment'] == 'multiple_languages' or task_data['environment'] == 'Data Science':
-            form = MultilangForm(task_data, task_fs)
-        elif task_data['environment'] == 'HDL':
-            form = HDLForm(task_data, task_fs)
-        elif task_data['environment'] == 'Notebook':
-            form = NotebookForm(task_data, task_fs)
-        else:
-            return
-
-        # Try to parse and validate all the information
-        try:
-            form.parse()
-            form.validate()
-        except InvalidGraderError as error:
-            return json.dumps({'status': 'error', 'message': error.message})
-
-        # Generate the grader
-        if form.task_data['generate_grader']:
-            form.generate_grader()
-            task_data['grader_test_cases'] = form.task_data['grader_test_cases']
+    if task_data['environment'] == 'multiple_languages' or task_data['environment'] == 'Data Science':
+        form = MultilangForm(task_data, task_fs)
+    elif task_data['environment'] == 'HDL':
+        form = HDLForm(task_data, task_fs)
+    elif task_data['environment'] == 'Notebook':
+        form = NotebookForm(task_data, task_fs)
+    else:
+        return
+    # Try to parse and validate all the information
+    try:
+        form.parse()
+        form.validate()
+    except InvalidGraderError as error:
+        return json.dumps({'status': 'error', 'message': error.message})
+
+    # Generate the grader
+    if form.task_data['generate_grader'] and task_data['generate_grader']:
+        form.generate_grader()
+        task_data['grader_test_cases'] = form.task_data['grader_test_cases']
+
+    if not task_data['generate_grader']:
+        #Keep track of test cases saved when generate_grader option is unchecked
+        task_data['grader_test_cases'] = form.task_data['grader_test_cases']
```

Comment:
Can we refactor this as follows? 

```suggestion
    if form.task_data['generate_grader'] and task_data['generate_grader']:
        form.generate_grader()
    task_data['grader_test_cases'] = form.task_data['grader_test_cases']
```

Reply:
Yes, works perfect.